### ISSUE TITLE
Initial Draft re-design [The Fede Edition]

### ISSUE THREAD

**Issue by schilkp** [schilkp/reginald#1]
0) Split input and data model

1) Add "Register Block" system:
```yaml
   CH1_SET: # Single register block instantiation
     start_adr: 0x01
     register_block_template: "CH_SETTINGS"

   CHn_SET: # Multiple register block instantiation
      start_adr: {1: 0x100, B: 0x200}
      replace_str_with_instance_name: "n" # Optional, Fallback: suffix w\o any padding char
      register_block_template: "CH_SETTINGS"
```
2) Move from set width to per-register with default:
```yaml
default_register_bitwidth: 8

...
registers:
  REGA:
    bitwidth: 8 # Override
```

3) Enum: 
     Refactor 'enums' to allow for brief+doc, and make syntax identical to field enum:
```yaml
register:
  A:
    ...
    fields:
       SETTINGS:
         bits: []
         brief:
         doc:
          enum: 
            VALA:
              ..
       OTHERFIELD:
         ...
         enum: GPIO_MODE
       

shared_enums:
  GPIO_MODE:
    brief:
    doc:
    enum:
      OUTPUT:
        value: 0
        brief: Pin configured as output.
      INPUT:
        value: 1
        brief: Pin configured as input.
```

4) Device Name -> Map name
```yaml
map_name: ADS129x
```
5) Field bit definitions:

```yaml
# Add:
    bits: 1 # == [1]
    bits: 1-4 # == [1,2,3,4]
    bits: [1, 4-10] # == [1,4,5,6,7,8,9,10]

# Remove: 
    bits:
      lsb_position: 1
      width: 3
```

6) Access:
```yaml
    access: [r, w]
```

**Comment by schilkp** [comment]
- c_funcpack.regs: 
     - pull addr defines to top
     - pull packing/unpacking to _reg_utils.h

**Comment by schilkp** [comment]
combine generics + regs into one file, with comments for register-specific packing/unpacking removeable using flag for large parts.

**Comment by schilkp** [comment]
rework reserved system:
remove 'reset_val'
add 'always_write'

RESP:
    adr: 0x16
    reset_val: 0x00
    brief: Respiration Control Register
    doc: This register provides the controls for the respiration circuitry
    always_write:
        bits: [8,0]
        value: [1, 1]  or 0x81 
    fields: ....



add 'modify' func for reg, which takes value, clears all fields and 'always_writes', and puts from struct into value.

pack falls back on modify, with a '0' base value.

**Comment by schilkp** [comment]
Implemented by cfd3a5e and b9cf9f4.
